### PR TITLE
Add base pyenv image for build and test purposes

### PR DESCRIPTION
**Pull request by paddycarey**
This PR adds a base image with pyenv and multiple versions of Python. This image is intended for use as a build and test environment for Python libraries.

### CODE REVIEW

**Comment by sk-jlewis** [comment]
I checked out this branch locally and ran it successfully.  Not sure if theres more things I could help to test before approving, but one feedback i could give is that it took about 9 minutes to build.  Is that acceptable?

**Comment by paddycarey** [comment]
Don't think there's much more to do than check that it seems sensible tbh.

> it took about 9 minutes to build. Is that acceptable?

I think so. I'm moving this into the dockerfiles repo from one of our internal libs so that we can avoid the build cost for each service and just use a prebuilt public image like this one. It'll still take a while to build when we push a new version, but for individual library builds it should just be a download, rather than build.

File: pyenv/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
+FROM ubuntu:18.04
```
Comment (sk-jlewis) [inline pyenv/Dockerfile]: Why was ubuntu chosen over an alpine image?

Comment (paddycarey) [reply pyenv/Dockerfile]: I couldn't convince pyenv to work properly on alpine, i'm sure it's possible just didn't want to sink too much time into it.

**Pull request merged**